### PR TITLE
Readd MCG CLI copying support for OCS <4.6

### PR DESCRIPTION
**Pull request by MeridianExplorer**
`tar`, which is needed for `oc cp`, was only added to the MCG operator pod in OCS 4.6.
This PR fixes it by mixing both the workaround method and `oc cp` based on OCS version.

Also added logging for extra clarity on `subprocess.run`, and used `exec_cmd()` for logging on `oc cp`

### CODE REVIEW

**Review by petr-balogh — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Comment by petr-balogh** [comment]
@MeridianExplorer When I ran with this PR I saw it failed here:
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/2605/testReport/tests.ecosystem.deployment/test_deployment/test_deployment/

```
                proc = subprocess.run(cmd, shell=True)
                logger.info(
>                   f"MCG CLI copying process stdout:{proc.stdout.decode()}, stderr: {proc.stderr.decode()}"
                )
E               AttributeError: 'NoneType' object has no attribute 'decode'
```

**Comment by petr-balogh** [comment]
As it has this issue I am adding WIP to title of PR to prevent it to be merged before we will find out why it's failed.
Thanks

**Comment by MeridianExplorer** [comment]
The problem stemmed form a missing `capture_output=True` which automatically initializes `stdout` and `stderr` to be `b''` even if they hold no data.
Verified locally.

**Comment by petr-balogh** [comment]
Running verification here:
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-trigger-vsphere-upi-1az-rhcos-vsan-3m-3w-upgrade-ocs-auto/137/

**Review by petr-balogh — APPROVED** [review]

File: ocs_ci/ocs/resources/mcg.py [hunk 1]
```diff
@@ -991,24 +992,46 @@ def _compare_cli_hashes():
             not os.path.isfile(constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH)
             or not _compare_cli_hashes()
         ):
+            logger.info(
+                f"The MCG CLI binary could not not found in {constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH},"
```
Comment (dahorak) [inline ocs_ci/ocs/resources/mcg.py:996]:
```suggestion
                f"The MCG CLI binary could not be found in {constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH},"
```

**Review by dahorak — APPROVED** [review]
Just small typo in log message, otherwise LGTM.

**Pull request merged**